mobile/src/main/java/org/openhab/habdroid/core/connection/ConnectionManagerHelper.kt

```diff
@@ -155,45 +148,7 @@ interface ConnectionManagerHelper {
         }
     }
 
-    @Suppress("DEPRECATION")
-    private class NetworkTypeHelperApi22 constructor(context: Context) {
-        private val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
-        val currentConnections: List<ConnectionType> get() {
-            return connectivityManager.allNetworkInfo
-                    .filter { info -> info.isConnected }
-                    .map { info -> when (info.type) {
-                        ConnectivityManager.TYPE_VPN -> ConnectionType.Vpn(null)
-                        ConnectivityManager.TYPE_WIFI -> ConnectionType.Wifi(null)
-                        ConnectivityManager.TYPE_BLUETOOTH -> ConnectionType.Bluetooth(null)
-                        ConnectivityManager.TYPE_ETHERNET -> ConnectionType.Ethernet(null)
-                        ConnectivityManager.TYPE_MOBILE -> ConnectionType.Mobile(null)
-                        else -> ConnectionType.Unknown(null)
-                    } }
-        }
-    }
-
-    @TargetApi(23)
-    @Suppress("DEPRECATION")
-    private class NetworkTypeHelperApi23 constructor(context: Context) {
-        private val connectivityManager = context.getSystemService(ConnectivityManager::class.java)!!
-        val currentConnections: List<ConnectionType> get() {
-            return connectivityManager.allNetworks
-                    .map { network -> network to connectivityManager.getNetworkInfo(network) }
-                    .filter { (_, info) -> info?.isConnected == true }
-                    // info can not be null here, as the condition in the line above covers that case already
-                    .map { (network, info) -> when (info!!.type) {
-                        ConnectivityManager.TYPE_VPN -> ConnectionType.Vpn(network)
-                        ConnectivityManager.TYPE_WIFI -> ConnectionType.Wifi(network)
-                        ConnectivityManager.TYPE_BLUETOOTH -> ConnectionType.Bluetooth(network)
-                        ConnectivityManager.TYPE_ETHERNET -> ConnectionType.Ethernet(network)
-                        ConnectivityManager.TYPE_MOBILE -> ConnectionType.Mobile(network)
-                        else -> ConnectionType.Unknown(network)
-                    } }
-        }
-    }
-
-    @TargetApi(26)
-    private class NetworkTypeHelperApi26 constructor(context: Context) {
+    private class NetworkTypeHelper constructor(context: Context) {
         private val connectivityManager = context.getSystemService(ConnectivityManager::class.java)!!
```

Comment:
AS: `Call requires API level 23 (current min is 21): android.content.Context#getSystemService`

Reply:
Weird that it didn't show that error for me, because it surely is one. Fixed now.